distribution/ddf-common/src/main/resources/bin/get_property

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+PROPERTY_NAME=$1
+
+# Useful directories
+SCRIPTDIR=$(dirname $0)
+HOME_DIR=$(cd "${SCRIPTDIR}/.."; pwd -P)
+
+# Useful files
+PROPERTIES_FILE=$HOME_DIR/etc/system.properties
+
+# Check file exists where it is supposed to
+if [ ! -f $PROPERTIES_FILE ]; then
+    echo Cannot find file $PROPERTIES_FILE >&2
+    exit 1
+fi
+
+# Extract the value of a property from a Java properties file
+# This function does not handle multi-line properties
+# If the property does not exist, print  error message to stderr
+value=""
+value=`grep ^$PROPERTY_NAME= "$PROPERTIES_FILE" | cut -d = -f2`
```

Comment:
✏️ Use `$(command | command)` instead of `` `command | command` ``

Reply:
Also I think this needs to be `grep "^$PROPERTY_NAME ="` since there is a space after the equals sign typically. If you want to handle both cases then don't search for the equals sign at all and you can later split the result like this:

```
match=$(grep "^$PROPERTY_NAME" "$PROPERTIES_FILE") # find match
IFS='=' read key value <<< "$match" # split match on = character
value=${value##*( )} # Trim whitespace. NOTE: this is bash specific, only suggested it because we are explicitly using bash in this script```

Reply:
Scratch that there isn't usually a space, however `some.property = "someValue"` is valid in a java properties file but will not be found by this as it currently stands. Definitely recommend the second approach

Reply:
That equals sign is not supposed to be there. It must be a copy paste error

Reply:
I would recommend not relying on bash builtins. I love the capabilities that bash provides, but have found that eventually someone will expect to be able to run shell scripts in an environment that only has /bin/sh available. In most modern operating systems /bin/bash can be expected to be available, but isn't a guarantee.

Reply:
I see. I will drop the equals match. The alternative is to match zero or more whitespace, but I don't' think the extra complexity really adds much value to the reg ex.